packages/react-aria-components/src/Checkbox.tsx

```diff
@@ -0,0 +1,241 @@
+  [props, ref] = useContextProps(props, ref, CheckboxGroupContext);
+  let state = useCheckboxGroupState(props);
+  let [labelRef, label] = useSlot();
+  let {groupProps, labelProps, descriptionProps, errorMessageProps} = useCheckboxGroup({
+    ...props,
+    label
+  }, state);
+
+  let renderProps = useRenderProps({
+    ...props,
+    values: {
+      isDisabled: state.isDisabled,
+      isReadOnly: state.isReadOnly,
+      isRequired: props.isRequired || false,
+      validationState: state.validationState
+    },
+    defaultClassName: 'react-aria-CheckboxGroup'
+  });
+
+  return (
+    <div 
+      {...groupProps}
+      {...renderProps}
+      ref={ref}
+      slot={props.slot}
+      data-readonly={state.isReadOnly || undefined}
+      data-required={props.isRequired || undefined}
+      data-validation-state={state.validationState || undefined}>
+      <Provider
+        values={[
+          [InternalCheckboxGroupContext, state],
+          [LabelContext, {...labelProps, ref: labelRef}],
+          [TextContext, {
+            slots: {
+              description: descriptionProps,
+              errorMessage: errorMessageProps
+            }
+          }]
+        ]}>
+        {renderProps.children}
+      </Provider>
+    </div>
+  );
+}
+
+export const CheckboxContext = createContext<ContextValue<CheckboxProps, HTMLInputElement>>(null);
+
+function Checkbox(props: CheckboxProps, ref: ForwardedRef<HTMLInputElement>) {
+  [props, ref] = useContextProps(props, ref, CheckboxContext);
+  let groupState = useContext(InternalCheckboxGroupContext);
+  let {inputProps, isSelected, isDisabled, isReadOnly, isPressed: isPressedKeyboard} = groupState
+    // eslint-disable-next-line react-hooks/rules-of-hooks
+    ? useCheckboxGroupItem({
+      ...props,
+      // Value is optional for standalone checkboxes, but required for CheckboxGroup items;
+      // it's passed explicitly here to avoid typescript error (requires ignore).
+      // @ts-ignore
+      value: props.value
+    }, groupState, ref)
+    // eslint-disable-next-line react-hooks/rules-of-hooks
+    : useCheckbox(props, useToggleState(props), ref);
+  let {isFocused, isFocusVisible, focusProps} = useFocusRing();
+  let isInteractionDisabled = isDisabled || isReadOnly;
+
+  // Handle press state for full label. Keyboard press state is returned by useCheckbox
+  // since it is handled on the <input> element itself.
+  let [isPressed, setPressed] = useState(false);
+  let {pressProps} = usePress({
+    isDisabled: isInteractionDisabled,
+    onPressStart(e) {
+      if (e.pointerType !== 'keyboard') {
+        setPressed(true);
+      }
+    },
+    onPressEnd(e) {
+      if (e.pointerType !== 'keyboard') {
+        setPressed(false);
+      }
+    }
+  });
+
+  let {hoverProps, isHovered} = useHover({
+    isDisabled: isInteractionDisabled
+  });
+
+  let pressed = isInteractionDisabled ? false : (isPressed || isPressedKeyboard);
```

Comment:
As mentioned in the comment above the usePress, the hooks only handle interactions on the input itself not on the label wrapper we added here

Reply:
right, but then i'm confused as to why it's not in our RSP implementation

Reply:
I think the DOM structure is different in RSP, so the input element actually extends to cover the whole label as well. So because of that we don't need any additional handlers. But here I cannot guarantee that the styling of the input would account for that. 
<img width="354" alt="image" src="https://user-images.githubusercontent.com/19409/223754805-00f76041-6a76-4379-a764-93f52f2e1979.png">

Reply:
ah, yep, i see that. i think we have a bug open about that actually, we could close it out with this work https://github.com/adobe/react-spectrum/issues/2383